docs/datatypes/stm.md

```diff
@@ -0,0 +1,115 @@
+---
+id: datatypes_stm
+title:  "STM"
+---
+
+Software Transactional Memory (STM) is a technique which allows composition of arbitrary atomic operations. It is the 
+software analog of transactions in database systems. STM can perform groups of memory operations in an atomic fashion. 
+The API is based on `Control.Concurrent.STM` although the implementation is completely different.
```

Comment:
Might want to mention this is in Haskell.

Reply:
done :)